plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/route/RouteChooser.java

```diff
@@ -55,6 +51,14 @@ public IRoute chooseRoute(List<IRoute> routes) {
 		return selectedRoute[0];
 	}
 
+	private void ensureHasShell() {
+		if (shell == null) {
+			shell = PlatformUI.getWorkbench()
+					.getActiveWorkbenchWindow()
+					.getShell();
+		}
+	}
+
```

Comment:
from comment above, this method could return the active workbench window's shell, to be used when necessary.